recipes/tinygrad/recipe.yaml

```diff
@@ -0,0 +1,137 @@
+    ${{ " or test_backward_sum_acc_dtype" if linux }}
+    ${{ "RuntimeError: Attribute list does not match Module context!" if 0 }}
+    ${{ " or test_bf16_disk_write_read" if linux and cuda_compiler_version != "None" }}
+    ${{ "`error: use of undefined value '@.const.pickledata.7893608704'`" if 0 }}
+    ${{ " or (TestWhisper and (test_transcribe_batch21 or test_transcribe_file1))" if osx }}
+    ${{ "signed int32 overflow" if 0 }}
+    ${{ " or test_float_midcast_int32" if osx }}
+    ${{ "tinygrad.codegen.kernel.KernelOptError: must have tensor cores or TC=2" if 0 }}
+    ${{ " or test_unmerged_ifs" if osx }}
+
+package:
+  name: tinygrad
+  version: ${{ version }}
+
+source:
+  url: https://github.com/tinygrad/tinygrad/archive/refs/tags/v${{ version }}.tar.gz
+  sha256: a22402a8c08da0930720b13a8a66a9c9f99507b15fca9caf31fde63776529ca8
+  patches:
+    - patches/0001-also-install-tinygrad.viz.patch
+    # clang 19 doesn't seem to support -std=metal3.1 yet
+    - patches/0002-downgrade-default-metal-standard.patch
+
+build:
+  number: ${{ build_number }}
+  string: ${{ torch_proc_type }}_py${{ python | version_to_buildstring }}_h${{ hash }}_${{ build_number }}
+  script: python -m pip install . -vv
+  skip:
+    - match(python, "<3.10")
+    # toolchain for 11.8 is not usable in end-user environment
+    - cuda_compiler_version == "11.8"
+    # skip windows until next version (where testing without jax should be possible),
+    # or when conda-forge gains windows-support for jax
+    - win
+    # only one CUDA build because there's not enough disk space to build all variants in one job
+    - cuda_compiler_version != "None" and match(python, "!=3.12")
+
+requirements:
+  host:
+    - python
+    - pip
+    - setuptools
+  run:
+    - python
+    - if: win
+      then:
+        - clang-cl_${{ target_platform }}
+      else:
+        - clangxx_${{ target_platform }}
+    - if: cuda_compiler_version != "None"
+      then:
+        - cuda-nvcc_${{ target_platform }}
+        - triton
+
+tests:
+  - python:
+      imports:
+      - tinygrad
+      pip_check: true
+  - requirements:
+      run:
+        # minimal
+        - pytest
+        - pytest-rerunfailures
+        - pytest-xdist
+        - hypothesis
+        - numpy
+        - pytorch
+        # optional
+        - blobfile
+        - bottle
+        - capstone
+        # https://github.com/conda-forge/staged-recipes/issues/29179
+        # - ggml-python
+        - if: unix
+          then:
+            # not available on windows yet
+            - jax
+        - librosa
+        - networkx
+        - nibabel
+        - onnx2torch
+        - onnx
+        - py-opencv
+        - pillow
+        - safetensors
+        - sentencepiece
+        - tabulate
+        - tiktoken
+        - tqdm
+        - transformers
+    files:
+      source:
+        - examples/
+        - extra/
+        - test/
```

Comment:
These are adding a buuunch of files to the package. Just mentioning it in case you want to think whether all of them are needed or not.

Reply:
I can split them off into a `tinygrad-tests` package, but I do want to keep running the test suite...

Reply:
Yea I think that's reasonable.